core/src/main/java/jenkins/util/AntWithFindResourceClassLoader.java

```diff
@@ -7,24 +7,23 @@
 import java.io.IOException;
 import java.lang.reflect.Field;
 import java.net.URL;
+import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Enumeration;
-import java.util.Vector;
 
 /**
  * As of 1.8.0, {@link org.apache.tools.ant.AntClassLoader} doesn't implement {@link #findResource(String)}
  * in any meaningful way, which breaks fast lookup. Implement it properly.
  */
 public class AntWithFindResourceClassLoader extends AntClassLoader implements Closeable {
-    private final Vector pathComponents;
+    private final ArrayList<File> pathComponents;
```

Comment:
This field is only used as a copy of parent's `pathComponents` (see the reflection code in constructor). That means `getUrl` is always called with the same first argument: if you remove that argument and use the field directly in `getUrl`, you can drop the copy of  the field in `AntWithFindResourceClassLoader`. Can be done in another PR though.